src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/message/recluster/ReclusterService.kt

```diff
@@ -34,22 +35,32 @@ class ReclusterService(
     val existingRecordsInCluster = cluster.personEntities.filterNot { it.id == changedRecord.id }
     val matchesToChangeRecord: List<PersonMatchResult> = personMatchService.findHighestConfidencePersonRecordsByProbabilityDesc(changedRecord)
     val matchedRecords: List<PersonEntity> = matchesToChangeRecord.map { it.personEntity }
+    val reclusterRelationship = ClusterRelationship(matchedRecords, existingRecordsInCluster)
     when {
-      recordsDontMatch(matchedRecords, existingRecordsInCluster) -> handleDiscrepancyOfMatchesToExistingRecords(matchedRecords, existingRecordsInCluster, cluster)
-      else -> telemetryService.trackEvent(CPR_RECLUSTER_NO_CHANGE, mapOf(EventKeys.UUID to cluster.personId.toString()))
+      reclusterRelationship.isDistinct() -> handleDiscrepancyOfMatchesToExistingRecords(reclusterRelationship, cluster)
+      else -> logNoChangeToCluster(cluster)
     }
   }
 
-  private fun handleDiscrepancyOfMatchesToExistingRecords(matchedRecords: List<PersonEntity>, existingRecordsInCluster: List<PersonEntity>, cluster: PersonKeyEntity) {
+  private fun handleDiscrepancyOfMatchesToExistingRecords(clusterRelationship: ClusterRelationship, cluster: PersonKeyEntity) {
     when {
-      hasLessMatchedRecordsInExistingCluster(matchedRecords, existingRecordsInCluster) -> setClusterAsNeedsAttention(cluster)
+      clusterRelationship.recordsInClusterNotMatched().isNotEmpty() -> handleLessMatchRecords(clusterRelationship.matchedRecords, cluster)
```

Comment:
not quite clear what has changed in the cluster - how about a method like

`clusterHasLostRecords`
`clusterIsSmaller`
`clusterHasShrunk`
`recordsHaveLeftCluster`